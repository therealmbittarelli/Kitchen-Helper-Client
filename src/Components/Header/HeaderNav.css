@font-face {
  font-family: 'cookiesregular';
  src: url('../../Assets/webfontkit-20200213-144022/cookies-webfont.woff') format('woff'),
       url('../../Assets/webfontkit-20200213-144022/cookies-webfont.woff2') format('woff2');
  font-weight: normal;
  font-style: normal;
}

h1 {
  font-family: 'cookiesregular';

}

#header-nav-container {
  text-align: center;
  text-decoration: none;
}

.heading{
  text-decoration: none;
  border: 3px solid black;
}

/* .log-out-user-link{
  margin-left: -400px;
} */

#site-sections-box-container {
  display: flex;
  flex-direction: row;
  justify-content: center;
}
.nav-link-text {
  margin: 10px;
  padding: 5px;
}

