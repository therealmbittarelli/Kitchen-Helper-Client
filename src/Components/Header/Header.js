import React, { Component } from "react";
import { Link } from "react-router-dom";
import TokenService from "../../Helpers/Token";
import UserContext from "../../Contexts/UserContext";
import LandingLogo from "../../Assets/LandingLogo.gif";
import "./HeaderNav.css";

class Header extends Component {
  static contextType = UserContext;

  handleLogoutClick = () => {
    this.context.processLogout();
  };

  renderLogoutLink() {
    return (
      <div>
        <nav className="log-out-user">
          <div className="logged-in-user">{this.context.user.name}</div>
          <Link
            className="log-out-user-link"
            onClick={this.handleLogoutClick}
            to="/login"
          >
            Logout
          </Link>
          {this.renderNavMenu()}
        </nav>
      </div>
    );
  }

  renderLoginLink() {
    return (
      <nav className="login-page-heading">
        <Link className="login-page-login login-page-link" to="/login">
          Login
        </Link>{" "}
        <Link className="login-page-register login-page-link" to="/register">
          Sign up
        </Link>
      </nav>
    );
  }

  renderNavMenu() {
    return (
      <section id="navBar">
        <Link className="dashboard-link-selector" to={"/home"}>
          Home
        </Link>
        <br />
        <Link className="nav-link-text pantry" to={"/pantry"}>
          Pantry
        </Link>
        <br />
        <Link className="nav-link-text recipes" to={"/recipes"}>
          Recipes
        </Link>
        <br />
        <Link className="nav-link-text meal-planning" to={"/planner"}>
          Meal Plans
        </Link>
      </section>
    );
  }

  render() {
    return (
      <header className="heading">
        <div className="landing-image">
          <a href="https://kitchen-helper-2020.now.sh/">
            <img src={LandingLogo} alt="Landing Logo" className="landinglogo" />
          </a>
          <Link className="login-page-link" to="/">

          </Link>
        </div>
        {TokenService.hasAuthToken()
          ? this.renderLogoutLink()
          : this.renderLoginLink()}
      </header>
    );
  }
}

export default Header;
