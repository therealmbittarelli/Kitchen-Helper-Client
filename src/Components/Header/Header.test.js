import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import { BrowserRouter } from 'react-router-dom';
import renderer from "react-test-renderer";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<BrowserRouter>
    <Header />
  </BrowserRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer.create(<BrowserRouter>
    <Header />
  </BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});