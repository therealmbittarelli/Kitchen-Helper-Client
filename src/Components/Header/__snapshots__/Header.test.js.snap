// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<header
  className="heading"
>
  <div
    className="landing-image"
  >
    <a
      href="http://localhost:3000/"
    >
      <img
        alt="Landing Logo"
        className="landinglogo"
        src="LandingLogo.gif"
      />
    </a>
    <a
      className="login-page-link"
      href="/"
      onClick={[Function]}
    />
  </div>
  <nav
    className="login-page-heading"
  >
    <a
      className="login-page-login login-page-link"
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
     
    <a
      className="login-page-register login-page-link"
      href="/register"
      onClick={[Function]}
    >
      Sign up
    </a>
  </nav>
</header>
`;
